VideoLocker/src/main/java/org/edx/mobile/third_party/iconify/IconDrawable.java

```diff
@@ -160,12 +177,25 @@ public int getIntrinsicWidth() {
 
     @Override
     public void draw(Canvas canvas) {
-        paint.setTextSize(getBounds().height());
+        Rect bounds = getBounds();
+        int height = bounds.height();
+        paint.setTextSize(height);
         Rect textBounds = new Rect();
         String textValue = valueOf(icon.character);
         paint.getTextBounds(textValue, 0, 1, textBounds);
-        float textBottom = (getBounds().height() - textBounds.height()) / 2f + textBounds.height() - textBounds.bottom;
-        canvas.drawText(textValue, getBounds().width() / 2f, textBottom, paint);
+        int textHeight = textBounds.height();
+        float textBottom = bounds.top + (height - textHeight) / 2f + textHeight - textBounds.bottom;
+        final boolean needMirroring = needMirroring();
+        if (needMirroring) {
+            canvas.save();
+            // Mirror the icon
+            canvas.translate(bounds.width(), 0);
+            canvas.scale(-1.0f, 1.0f);
+        }
+        canvas.drawText(textValue, bounds.exactCenterX(), textBottom, paint);
+        if (needMirroring) {
+            canvas.restore();
+        }
```

Comment:
@1zaman  that means each time we need to redraw it.  it is expensive when we use it in list view.